plugins/module_utils/common.py

```diff
@@ -249,16 +264,34 @@ def find_resource(self, kind, api_version, fail=False):
             if fail:
                 self.fail(msg='Failed to find exact match for {0}.{1} by [kind, name, singularName, shortNames]'.format(api_version, kind))
 
-    def kubernetes_facts(self, kind, api_version, name=None, namespace=None, label_selectors=None, field_selectors=None):
+    def kubernetes_facts(self, kind, api_version, name=None, namespace=None, label_selectors=None, field_selectors=None,
+                         wait=False, wait_sleep=5, wait_timeout=120, state='present', condition=None):
         resource = self.find_resource(kind, api_version)
         if not resource:
             return dict(resources=[])
+
+        if not label_selectors:
+            label_selectors = []
+        if not field_selectors:
+            field_selectors = []
+
         try:
             result = resource.get(name=name,
                                   namespace=namespace,
                                   label_selector=','.join(label_selectors),
-                                  field_selector=','.join(field_selectors)).to_dict()
-        except openshift.dynamic.exceptions.NotFoundError:
+                                  field_selector=','.join(field_selectors))
+            satisfied_by = []
+            if wait and result['kind'].endswith('List') and hasattr(result, 'items'):
+                # We have a list of ResourceInstance
+                for resource_instance in result.get('items'):
+                    success, res, duration = self.wait(resource, resource_instance, sleep=wait_sleep, timeout=wait_timeout)
+                    if not success:
+                        self.fail(msg="Failed to gather information about %s(s)" % result.get('kind')[:-4])
```

Comment:
`result.base_kind` should have the proper kind in it if you want to avoid the string manipulation

Reply:
`result` is `ResourceInstance` and does not have `base_kind` (which is for ResourceList).

`res` is a dictionary since `self.wait` returns the dictionary.

Reply:
yep ignore all the comments about ResourceLists